src/theory/theory.h

```diff
@@ -77,11 +77,34 @@ namespace eq {
  * RegisteredAttr works.  (If you need multiple instances of the same
  * theory, you'll have to write a multiplexed theory that dispatches
  * all calls to them.)
+ *
+ * NOTE: A Theory has a special way of being initialized. The owner of a Theory
+ * is either:
+ *
+ * (A) Using Theory as a standalone object, not associated with a TheoryEngine.
+ * In this case, simply call the public initialization method
+ * (Theory::finishInitStandalone).
+ *
+ * (B) TheoryEngine, which determines how the Theory acts in accordance with
+ * its theory combination policy. We require the following steps in order:
+ * (B.1) Get information about which equality engine notifications the Theory
+ * would like to be notified of (Theory::needsEqualityEngine).
```

Comment:
Wouldn't it be more accurate to say something like:

> (B.1) Get information about whether this Theory requires an equality engine for congruence-based reasoning (Theory::needsEqualityEngine).

Reply:
It will be both, whether we want to maintain one, and which notifications we want.  I'll update the comment.